src/Combinatorics/Matroids/matroids.jl

```diff
@@ -0,0 +1,825 @@
+    if check && length(groundset)!=length(Set(groundset))
+        error("Input is not a valid groundset of a matroid")
+    end
+    gs2num = create_gs2num(groundset)
+    pm_hyperplanes = [[gs2num[i]-1 for i in H] for H in hyperplanes]
+    M = Polymake.matroid.Matroid(MATROID_HYPERPLANES=pm_hyperplanes,N_ELEMENTS=length(groundset))
+    #TODO implement a check if these are actually the hyperplanes of a matroid
+    return Matroid(M,groundset,gs2num)
+end
+
+@doc Markdown.doc"""
+    matroid_from_matrix_columns(A::MatrixElem)
+
+A matroid represented by the column vectors of a matrix `A`.
+
+See Section 1.1 of [Oxl11](@cite).
+
+# Examples
+To construct the vector matroid (a.k.a linear matroid) of the matrix `A` over the field with two elements write:
+```jldoctest
+julia> A = matrix(GF(2),[[1,0,1,1],[0,1,1,1]]);
+
+julia> M = matroid_from_matrix_columns(A)
+Matroid of rank 2 on 4 elements
+```
+"""
+function matroid_from_matrix_columns(A::MatrixElem; check::Bool=true)
+    rk = rank(A)
+    nr = nrows(A)
+    bases = Vector{Vector{Int64}}()
+    # We use a temporary matrix which we will rewrite to avoid creating new matrices in every iteration.
+    tmp_mat = A[:,1:rk]
+    for set in subsets(Vector(1:ncols(A)),rk)
+        for is in 1:rk
+            for ir in 1:nr
+                tmp_mat[ir,is] = A[ir,set[is]]
+            end
+        end
+        if rank(tmp_mat)==rk
+            push!(bases, set);
+        end
+    end
+    return matroid_from_bases(bases, ncols(A); check=check)
+end
+
+@doc Markdown.doc"""
+    matroid_from_matrix_columns(A::MatrixElem)
+
+A matroid represented by the row vectors of a matrix.
+
+See Section 1.1 of [Oxl11](@cite).
+
+# Examples
+To construct the linear matroid of the rows of the matrix `A` over the field with two elements write:
+```jldoctest
+julia> A = matrix(GF(2),[[1,0],[0,1],[1,1],[1,1]]);
+
+julia> M = matroid_from_matrix_rows(A)
+Matroid of rank 2 on 4 elements
+```
+"""
+matroid_from_matrix_rows(A::MatrixElem, ; check::Bool=true) = matroid_from_matrix_columns(transpose(A); check=check)
+
+@doc Markdown.doc"""
+    cycle_matroid(g::Oscar.Graphs.Graph)
+
+The cycle matroid of a graph `g`.
+
+See Section 1.1 of [Oxl11](@cite).
+
+# Examples
+To construct the cycle matroid of the complete graph of 4 vertices write:
+```jldoctest
+julia> g = Oscar.Graphs.complete_graph(4);
+
+julia> M = cycle_matroid(g)
+Matroid of rank 3 on 6 elements
+```
+"""
+function cycle_matroid(g::Oscar.Graphs.Graph)
+    pm_Graph = Polymake.graph.Graph(ADJACENCY=g.pm_graph)
+    M = Polymake.matroid.matroid_from_graph(pm_Graph)
+    n = Oscar.Graphs.ne(g)
+    gs2num = create_gs2num(Vector{Int}(1:n))
+    return Matroid(M,1:n,gs2num)
+end
+
+@doc Markdown.doc"""
+    bond_matroid(g::Oscar.Graphs.Graph)
+
+The `bond matroid` or `cocylce matroid` of a graph `g` which is the dual of a cycle matroid, e.g, cographic.
```

Comment:
Backticks are for code, not quotes. Also: typos/
```suggestion
The "bond matroid" or "cocycle matroid" of a graph `g` which is the dual of a cycle matroid, e.g. cographic.
```
I also don't understand the example, i.e., the bit after "e.g.": is this meant to say "a cographic matroid is a cycle matroid" ? Or is it meant to say that a "bond matroid" is cographic? Or what else?

Reply:
Both a "bond matroid" is cographic and a cographic matroid is a cycle matroid (of another graph)